include/rdma/providers/fi_peer.h

```diff
@@ -181,7 +182,8 @@ struct fi_ops_srx_owner {
 	int	(*get_msg)(struct fid_peer_srx *srx, fi_addr_t addr,
 			size_t size, struct fi_peer_rx_entry **entry);
 	int	(*get_tag)(struct fid_peer_srx *srx, fi_addr_t addr,
-			size_t size, uint64_t tag, struct fi_peer_rx_entry **entry);
+			size_t size, uint64_t tag, uint64_t cq_data,
+			uint64_t flags, struct fi_peer_rx_entry **entry);
```

Comment:
cq_data and flags are completion data.  It seems odd to pass that into a call that is trying to retrieve the buffer.

I wonder if util_init_unexp() is being called from the correct side.  There's a mismatch here setting some of the fi_peer_rx_entry fields based on whether we matched a buffer or not that seems confusing.  The size, cq_data, and flags fields are ignored unless we don't find a match?

Reply:
yes I agree, it seems much. Right now the definition of the peer entry is that only the owner is allowed to touch the fields (hence this change). But the other alternative is to modify the definition so that the peer is allowed to initialize all the variables only in the unexpected case. I think I like that better. Thoughts?

Reply:
Because we have a separate queue call, it should be okay for the peer to modify the structure prior to queuing it.